### PR TITLE
Voice providers part 1

### PR DESCRIPTION
**Pull request by snopoke**
Data models and views for managing voice providers for a team.

PR on top of https://github.com/dimagi/open-chat-studio/pull/87

### CODE REVIEW

File: apps/service_providers/tables.py [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
 
-    class Meta:
-        model = LlmProvider
-        fields = (
-            "type",
-            "name",
-        )
-        row_attrs = settings.DJANGO_TABLES2_ROW_ATTRS
-        orderable = False
+    class_name = model.__name__ + "AutogeneratedTable"
```
Comment (SmittieC) [inline apps/service_providers/tables.py:16]: This is cool

File: apps/service_providers/utils.py [hunk 2]
```diff
@@ -1,36 +1,88 @@
...
-        form.fields["type"].disabled = True
+        form.fields[provider.secondary_key_field].disabled = True
 
     return form
 
 
-def formfield_for_dbfield(db_field, **kwargs):
-    if db_field.name == "type":
-        return forms.TypedChoiceField(empty_value=None, choices=LlmProviderType.choices)
+def formfield_for_dbfield(db_field, provider, **kwargs):
```
Comment (SmittieC) [inline apps/service_providers/utils.py:84]: nit: can we add a type hints here?

File: apps/service_providers/views.py [hunk 3]
```diff
@@ -3,35 +3,45 @@
...
-from .tables import LlmProviderTable
-from .utils import get_llm_config_form
+from .utils import ServiceProvider, get_service_provider_config_form
 
 
-class LlmProviderTableView(SingleTableView):
+class ServiceProviderMixin:
+    @property
+    def provider_type(self) -> ServiceProvider:
+        type_ = self.kwargs["provider_type"]
```
Comment (SmittieC) [inline apps/service_providers/views.py:12]: Where does `self.kwargs` come from?

**Review by SmittieC — APPROVED** [review]
This was brain stretching, thanks. I wonder if a small `Readme` in this app on how to add more service models might not be a good investment. I can see us adding for instance a transcription service as well in the future

**Comment by snopoke** [comment]
> This was brain stretching, thanks. I wonder if a small `Readme` in this app on how to add more service models/views might not be a good investment. I can see us adding for instance a transcription service as well in the future

I noticed that we already do transcription, I was planning to reuse the LlmProvider for now.

File: apps/service_providers/views.py (hunk 3 above)
Comment (snopoke) [reply apps/service_providers/views.py:12]: They are the standard Django view kwargs.

**Pull request merged**